receive.c

```diff
@@ -391,13 +436,14 @@ int main(int argc, char **argv) {
 		}
 	}
 
-	if(filteropt == 1) {
+	if(filtertype > 0) {
 		struct protocol_t *protocol = NULL;
-		nrfilter = explode(filter, ",", &filters);
 		unsigned int match = 0, j = 0;
 
 		protocol_init();
 
+		nrfilter = explode(filter, ",", &filters);
```

Comment:
Found it, it uses the same free as the previous implementation: https://github.com/pilight/pilight/pull/367/files#diff-eb620f32a6e8e7ceb3963dd45fb70665R59

Somehow this was already present in the rewrite code, so it's not shown in the diff.